src/content_mapblock.cpp

```diff
@@ -30,6 +30,27 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "log.h"
 #include "noise.h"
 
+// Distance of light extrapolation (for oversized nodes)
+// After this distance, it gives up and considers light level constant
+#define SMOOTH_LIGHTING_OVERSIZE 1.0
```

Comment:
I'm wondering if oversize extrapolation is worth doing, since it will probably not match lighting anyway and not doing this may look just as good. Not many nodes are oversized.